mkdirp.js

```diff
@@ -5,18 +5,20 @@ var path = require('path');
 var fs = require('graceful-fs');
 
 var MASK_MODE = parseInt('7777', 8);
-var DEFAULT_DIR_MODE = parseInt('0777', 8);
 
-function mkdirp(dirpath, customMode, callback) {
-  if (typeof customMode === 'function') {
-    callback = customMode;
-    customMode = undefined;
+function mkdirp(dirpath, mode, callback) {
+  if (typeof mode === 'function') {
+    callback = mode;
+    mode = undefined;
+  }
+
+  if (typeof mode === 'string') {
+    mode = parseInt(mode, 8);
   }
 
-  var mode = customMode || DEFAULT_DIR_MODE & ~process.umask();
   dirpath = path.resolve(dirpath);
 
-  fs.mkdir(dirpath, mode, onMkdir);
+  fs.mkdir(dirpath, { mode: mode }, onMkdir);
```

Comment:
The options object form of `fs.mkdir` requires node.js v10.12.0.  If you want to be compatible with node.js 10.0.0 then this needs to switch back to `fs.mkdir(dirname, mode, onMkdir)`.  If not I'd lean towards saying that `package.json#engines.node` should specify `>=10.12.0`.

Other than this I think LGTM!

Reply:
Good catch! I'll put that back.